src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -329,10 +329,16 @@ TransactionManager::update_lba_mappings(
         // for rewritten extents, last_committed_crc should have been set
         // because the crc of the original extent may be reused.
         // also see rewrite_logical_extent()
-        if (!extent->get_last_committed_crc()) {
-          extent->set_last_committed_crc(extent->calc_crc32c());
-        }
-        assert(extent->calc_crc32c() == extent->get_last_committed_crc());
+	if (!extent->get_last_committed_crc()) {
+	  if (epm->get_checksum_needed(extent->get_paddr())) {
+	    extent->set_last_committed_crc(extent->calc_crc32c());
+	  } else {
+	    extent->set_last_committed_crc(CRC_NULL);
+	  }
+	}
+	if (epm->get_checksum_needed(extent->get_paddr())) {
+	  assert(extent->calc_crc32c() == extent->get_last_committed_crc());
+	}
```

Comment:
```c++
#ifndef NDEBUG
if (epm->get_checksum_needed(extent->get_paddr())) {
  assert(extent->get_last_committed_crc() == extent->calc_crc32c());
} else {
  assert(extent->get_last_committed_crc() == CRC_NULL);
}
#endif
```
The ifndef is useful because in the release build `get_checksum_needed()` doesn't need to be called here.